In [1]:
import pandas as pd

In [21]:
pd.set_option('max_column', None)
df = pd.read_excel('Week1_challenge_data_source.xlsx')
df.head()

Bearer Id               Start  Start ms                 End  \
0  13114483460844900352 2019-04-04 12:01:18     770.0 2019-04-25 14:35:31   
1  13114483482878900224 2019-04-09 13:04:04     235.0 2019-04-25 08:15:48   
2  13114483484080500736 2019-04-09 17:42:11       1.0 2019-04-25 11:58:13   
3  13114483485442799616 2019-04-10 00:31:25     486.0 2019-04-25 07:36:35   
4  13114483499480700928 2019-04-12 20:10:23     565.0 2019-04-25 10:40:32   

   End ms  Dur. (ms)          IMSI  MSISDN/Number          IMEI  \
0   662.0  1823652.0  2.082014e+14   3.366496e+10  3.552121e+13   
1   606.0  1365104.0  2.082019e+14   3.368185e+10  3.579401e+13   
2   652.0  1361762.0  2.082003e+14   3.376063e+10  3.528151e+13   
3   171.0  1321509.0  2.082014e+14   3.375034e+10  3.535661e+13   
4   954.0  1089009.0  2.082014e+14   3.369980e+10  3.540701e+13   

  Last Location Name  Avg RTT DL (ms)  Avg RTT UL (ms)  \
0   9164566995485190             42.0              5.0   
1            L77566A             65.0              5.0   
2            D42335A              NaN              NaN   
3            T21824A              NaN              NaN   
4            D88865A              NaN              NaN   

   Avg Bearer TP DL (kbps)  Avg Bearer TP UL (kbps)  \
0                     23.0                     44.0   
1                     16.0                     26.0   
2                      6.0                      9.0   
3                     44.0                     44.0   
4                      6.0                      9.0   

   TCP DL Retrans. Vol (Bytes)  TCP UL Retrans. Vol (Bytes)  \
0                          NaN                          NaN   
1                          NaN                          NaN   
2                          NaN                          NaN   
3                          NaN                          NaN   
4                          NaN                          NaN   

   DL TP < 50 Kbps (%)  50 Kbps < DL TP < 250 Kbps (%)  \
0                100.0                             0.0   
1                100.0                             0.0   
2                100.0                             0.0   
3                100.0                             0.0   
4                100.0                             0.0   

   250 Kbps < DL TP < 1 Mbps (%)  DL TP > 1 Mbps (%)  UL TP < 10 Kbps (%)  \
0                            0.0                 0.0                100.0   
1                            0.0                 0.0                100.0   
2                            0.0                 0.0                100.0   
3                            0.0                 0.0                100.0   
4                            0.0                 0.0                100.0   

   10 Kbps < UL TP < 50 Kbps (%)  50 Kbps < UL TP < 300 Kbps (%)  \
0                            0.0                             0.0   
1                            0.0                             0.0   
2                            0.0                             0.0   
3                            0.0                             0.0   
4                            0.0                             0.0   

   UL TP > 300 Kbps (%)  HTTP DL (Bytes)  HTTP UL (Bytes)  \
0                   0.0              NaN              NaN   
1                   0.0              NaN              NaN   
2                   0.0              NaN              NaN   
3                   0.0              NaN              NaN   
4                   0.0              NaN              NaN   

   Activity Duration DL (ms)  Activity Duration UL (ms)   Dur. (ms).1  \
0                    37624.0                    38787.0  1.823653e+09   
1                      168.0                     3560.0  1.365104e+09   
2                        0.0                        0.0  1.361763e+09   
3                     3330.0                    37882.0  1.321510e+09   
4                        0.0                        0.0  1.089009e+09   

  Handset Manufacturer                 Handset Type  \
0      

In [20]:
# Reading in and printing out description of all columns in the dataset
field_desc = pd.read_excel('Field_Descriptions.xlsx')
count = 0
for i in field_desc['Fields']:
    print(f'{i}: {field_desc.Description[count]}\n')
    count += 1

bearer id: xDr session identifier

Dur. (ms): Total Duration of the xDR (in ms)

Start: Start time of the xDR (first frame timestamp)

Start ms: Milliseconds offset of start time for the xDR (first frame timestamp)

End: End time of the xDR (last frame timestamp)

End ms: Milliseconds offset of end time of the xDR (last frame timestamp)

Dur. (s): Total Duration of the xDR (in s)

IMSI: International Mobile Subscriber Identity

MSISDN/Number: MS International PSTN/ISDN Number of mobile - customer number

IMEI: International Mobile Equipment Identity

Last Location Name: User location call name (2G/3G/4G) at the end of the bearer

Avg RTT DL (ms): Average Round Trip Time measurement Downlink direction (msecond)

Avg RTT UL (ms): Average Round Trip Time measurement Uplink direction (msecond)

Avg Bearer TP DL (kbps): Average Bearer Throughput for Downlink (kbps) - based on BDR duration

Avg Bearer TP UL (kbps): Average Bearer Throughput for uplink (kbps) - based on BDR duration

TCP DL R

In [23]:
# Checking the shape of the dataset
df.shape

(150001, 55)

The dataset has 150001 rows and 55 columns

### Sub-Tasks

In [ ]:
# 